beacon-chain/rpc/beacon/attestations.go

```diff
@@ -127,25 +108,35 @@ func (bs *Server) ListAttestations(
 func (bs *Server) ListIndexedAttestations(
 	ctx context.Context, req *ethpb.ListIndexedAttestationsRequest,
 ) (*ethpb.ListIndexedAttestationsResponse, error) {
-	atts := make([]*ethpb.Attestation, 0)
+	blocks := make([]*ethpb.SignedBeaconBlock, 0)
 	var err error
 	epoch := helpers.SlotToEpoch(bs.GenesisTimeFetcher.CurrentSlot())
 	switch q := req.QueryFilter.(type) {
-	case *ethpb.ListIndexedAttestationsRequest_TargetEpoch:
-		atts, err = bs.BeaconDB.Attestations(ctx, filters.NewFilter().SetTargetEpoch(q.TargetEpoch))
+	case *ethpb.ListIndexedAttestationsRequest_GenesisEpoch:
+		blocks, err = bs.BeaconDB.Blocks(ctx, filters.NewFilter().SetStartEpoch(0).SetEndEpoch(0))
 		if err != nil {
 			return nil, status.Errorf(codes.Internal, "Could not fetch attestations: %v", err)
 		}
-		epoch = q.TargetEpoch
-	case *ethpb.ListIndexedAttestationsRequest_GenesisEpoch:
-		atts, err = bs.BeaconDB.Attestations(ctx, filters.NewFilter().SetTargetEpoch(0))
+	case *ethpb.ListIndexedAttestationsRequest_Epoch:
+		blocks, err = bs.BeaconDB.Blocks(ctx, filters.NewFilter().SetStartEpoch(q.Epoch).SetEndEpoch(q.Epoch))
 		if err != nil {
 			return nil, status.Errorf(codes.Internal, "Could not fetch attestations: %v", err)
 		}
-		epoch = 0
 	default:
 		return nil, status.Error(codes.InvalidArgument, "Must specify a filter criteria for fetching attestations")
 	}
+	totalLen := 0
+	for _, block := range blocks {
```

Comment:
if this way is the best way, please refactor into its own function

Reply:
Good idea, will try the benchmark.